### ISSUE TITLE
Is it an oversight that QueryCommandBuilder doesn't support include statements like BrowseCommandBuilder?

### ISSUE THREAD

**Issue by judgeaxl** [dipscope/EntityStore.TS#12]
When browsing using a filter, I can add include statements, like:

```
  myEntityStore.mySet
    .filter((u, f) => f.eq(u.id, some_id))
    .include(u => u.some_relation)
    .findAll()
```

But there doesn't seem to be a way to do the same for queries, these both complain there's no such function.

```
  myEntityStore.mySet
    .find(some_id)
    .include(u => u.some_relation) // <-- NOT A FUNCTION
```

```
  myEntityStore.mySet
    .include(u => u.some_relation)
    .find(some_id) // <-- NOT A FUNCTION
```

Is there any particular reason for this omission, am I using it wrong, or it's just something waiting to be implemented?

**Comment by dpimonov** [comment]
Hi, thanks for the question. Yeah, this is an oversight and definitely should be implemented. For now you can workaround it by doing the following. 

```
myEntityStore.mySet
    .filter((u, f) => f.eq(u.id, some_id))
    .include(u => u.some_relation)
    .findOne();
```

Not the best solution from performance perspective but a valid one. If you are using a pagination api - you can also limit the query with 1 record. Are you using EntityStore as a standalone component with a custom implementation or with `JsonApi` plugin?

Are there any other improvements you would like to see in the library? I am going to provide an update for a core package for working with types soon. After that I can consider the required changes to implement a missing feature. 

If you have time and desire - contribution is always welcome. :)

**Comment by judgeaxl** [comment]
For now I'm using the EntityStore server side in an Next.js application to interact with an external JsonAPI service, so I'm using the JsonApi entity provider with it.

I was doing the filter and `findOne()` solution to overcome this, but I realized the server would return slightly different data and until I could solve that I figured `find()` would do the trick. I'm still working on the overall app architecture so this exposed some shortcomings in the external service as well.

I've ran into a couple of small issues but primarily with JsonApi having a few assumptions about the server's implementation of filters, but I can note those down as suggestions for improvements in that repo.

Other than that, my biggest issue was in figuring out how to annotate the types without using the @Type and @Property decorators, especially for collections and generics. Some more documentation on that would be great.

Another issue I ran into was that I realized I don't actually need the type conversion most of the time, since I still have to serialize everything back to JSON to get it from the server to the client when using API routes, and then I had to override the naming convention or things ended up really confusing. I can explain more about that if you're interested. It's not a shortcoming of the library itself I'd say, more perhaps an unexpected use case.

**Comment by dpimonov** [comment]
Thanks for the feedback. It is really helpful to know about use cases other then mine.

```
Other than that, my biggest issue was in figuring out how to annotate the types without using the @type and https://github.com/Property decorators, especially for collections and generics. Some more documentation on that would be great.
```

This is a topic I am working on currently. It will be covered with an update I mentioned above. In short there will be a fluent API for configuration without decorators. It was not documented well as it was a point to change and improve from the start. Good to know that there is a real request for this feature. Hope to release that update soon but it depends from free time.

```
Another issue I ran into was that I realized I don't actually need the type conversion most of the time, since I still have to serialize everything back to JSON to get it from the server to the client when using API routes, and then I had to override the naming convention or things ended up really confusing. I can explain more about that if you're interested. It's not a shortcoming of the library itself I'd say, more perhaps an unexpected use case.
```

Details and examples will be useful so if you have time to describe in details I will appreciate.

**Comment by dpimonov** [comment]
Improvement is implemented in v2.0.0.
